### PR TITLE
[incubator/jaeger] Adding cassandra latest chart and version 3.11.3

### LINKED ISSUE #9696: [incubator/jaeger] Latest cassandra versions in requirements.yaml
**Is this a request for help?**:
No
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): FEATURE REQUEST

Upstream cassandra chart on `incubator/cassandra` has moved many versions ahead with many additional useful features like ability specify tolerations. 
`requirements.yaml` in `incubator/jaeger` although points to a pretty old version. 
https://github.com/helm/charts/blob/master/incubator/cassandra/Chart.yaml#L2

https://github.com/helm/charts/blob/master/incubator/jaeger/requirements.yaml#L3

**Version of Helm and Kubernetes**:
Helm : 2.11.0
Kubernetes: 1.10.6

**Which chart**: incubator/jaeger

**What happened**: Chart uses old dependency for cassandra

**What you expected to happen**: Use newer cassandra release (0.9.6) whereas current requirements.yaml uses 0.1.6 which is almost a year old. 
 

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by harshal-shah**
Signed-off-by: Harshal Shah <harshal2623@gmail.com>

#### What this PR does / why we need it:
This PR adds latest cassandra chart to requirements.yaml
Also it includes a minor fix for new URL of incubator charts. 
#### Which issue this PR fixes
  - fixes #9696 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: incubator/jaeger/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: 1.4.1
```
Comment (cpanato) [inline incubator/jaeger/Chart.yaml]: since you are updating the jaeger image as well you need to update here to 1.8.2

Comment (harshal-shah) [reply incubator/jaeger/Chart.yaml]: yes. Thanks for the review. I will send a new update.

Comment (harshal-shah) [reply incubator/jaeger/Chart.yaml]: While we are on it, any suggestions on why lint is failing? I'm able to add incubator charts successfully via `helm repo add incubator http://storage.googleapis.com/kubernetes-charts-incubator` but the CI does not like it

Comment (cpanato) [reply incubator/jaeger/Chart.yaml]: lets see if this happens again when you push the update

**Pull request merged**